### PR TITLE
gh-109599: Add types.CapsuleType

### PR DESCRIPTION
**Pull request by pitrou**
* Issue: gh-109599

----
:books: Documentation preview :books:: https://cpython-previews--109600.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/types.rst [hunk 1]
```diff
@@ -472,6 +472,12 @@ Standard names are defined for the following types:
 
       .. versionadded:: 3.12
 
+.. class:: CapsuleType
+
+   The type of :ref:`capsule objects<capsules>`.
```
Comment (serhiy-storchaka) [inline Doc/library/types.rst]: Should there be a space before `<`? I just never seen it without a space.

Comment (pitrou) [reply Doc/library/types.rst]: There is one occurrence just above, but I can add a space for consistency:
```restructuredtext
.. class:: UnionType

   The type of :ref:`union type expressions<types-union>`.

   .. versionadded:: 3.10
```

Comment (AlexWaygood) [reply Doc/library/types.rst]: > Should there be a space before `<`? I just never seen it without a space.

Both are fine as far as Sphinx is concerned, but in CPython's docs we nearly always add a space, for readability reasons

**Comment by pitrou** [comment]
@jorisvandenbossche @lidavidm Do you think this would be a useful addition (for example for typing annotations, unit tests, introspection...)?

**Comment by lidavidm** [comment]
I think it would be useful to let us be precise/explicit (I liked the original proposal of something like `Capsule["dlpack"]` best, though!). It could also be useful for writing stub files for C extensions/Cython code.

**Pull request merged**